plutus-metatheory/src/Type/BetaNBE/SoundnessRed.lagda

```diff
@@ -1,241 +0,0 @@
-\begin{code}
-module Type.BetaNBE.SoundnessRed where
```

Comment:
I think this is an experiment in proving properties about normalisation with respect to reduction rather than beta-eta-equality. 

No need to keep it unless we actually want to switch to that. Also Type.ReductionC/S, Type.CC and Type.CK are not used either so could be removed. I implemented Type.CC/CK and equivalence proofs as a warmup to doing it for terms.

Reply:
Then I'll erase: 
- `Type.ReductionS`
- `Type.ReductionC`
- `Type.CC`
- `Type.CK`

And update the `index.lagda.md` file accordingly.